### PR TITLE
chore: Migrate from AWS SDKv2 to v3

### PR DESCRIPTION
**Pull request by fredex42**
## What does this change?

- Migrates all usages of AWS SDK v2 to v3
- v3 is bundled with the NodeJS 18.x lambda runtime, so we can drop the bundling of the SDK library

## How to test

Deploy to a single test account and do a test manual rotation on a CODE stack as per the README instructions

## How can we measure success?

No more deprecation errors

## Have we considered potential risks?

Provided that this runs correctly, there should not be any extra risks associated with it

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,11 @@
 {
   "name": "elasticsearch-node-rotation",
   "dependencies": {
-    "aws-sdk": "^2.1142.0"
+    "@aws-sdk/client-auto-scaling": "^3.405.0",
+    "@aws-sdk/client-ec2": "^3.406.0",
```
Comment (akash1810) [inline package.json:5]: Any reason this is a different version from other `@aws-sdk` libraries?

**Review by akash1810 — APPROVED** [review]
Thanks for this!

I've [deployed](https://riffraff.gutools.co.uk/deployment/view/c6766d92-94cd-42a0-b04b-c414ab39471b) this to the DevX account, and started a rotation of the Central ELK nodes.

The shard migration step takes a little while to complete, so will not wait for the step function to fully complete before approving.

File: package.json (hunk 1 above)
Comment (fredex42) [reply package.json:5]: ooh, good spot. I think that must be an accident relating to the fact i installed them with `npm` and didn't do them all at the same time

Comment (fredex42) [reply package.json]: I've bumped them all to 3.408.0 as that's the latest version

**Pull request merged**